src/datasources/alarms/AlarmsDataSourceCore.ts

```diff
@@ -143,7 +143,7 @@ export abstract class AlarmsDataSourceCore extends DataSourceBase<AlarmsQuery> {
   }
 
   public shouldRunQuery(query: AlarmsQuery): boolean {
-    return !query.hide;
+    return !query?.hide;
```

Comment:
Would there be a problem if `query` was `undefined`/`null` and therefore `shouldRunQuery` returned `true`? Might be better to do `!(query?.hide ?? true)` or `query && !query.hide`.

Reply:
@OliverCosma replied on the linked thread saying the defensiveness of the check is not intentional / for a known issue so I'll remove it